sdks/wasm/src/library_mono.js

```diff
@@ -150,26 +171,33 @@ var MonoSupportLib = {
 			var ptr = Module._malloc(numBytes);
 			var heapBytes = new Int32Array(Module.HEAP32.buffer, ptr, numBytes);
 			for (let i=0; i<var_list.length; i++) {
-				heapBytes[i] = var_list[i]
+				heapBytes[i] = var_list[i][0]
 			}
 
 			this._async_method_objectId = 0;
 			this.mono_wasm_get_var_info (scope, heapBytes.byteOffset, var_list.length);
 			Module._free(heapBytes.byteOffset);
 			var res = MONO._fixup_name_value_objects (this.var_info);
 
-			//Async methods are special in the way that local variables can be lifted to generated class fields
-			//value of "this" comes here either
 			for (let i in res) {
-				var name = res [i].name;
-				if (name != undefined && name.indexOf ('>') > 0)
-					res [i].name = name.substring (1, name.indexOf ('>'));
-			}
+				var res_name = res [i].name;
+
+				var value = res[i].value;
+				if (this._async_method_objectId != 0) {
+					//Async methods are special in the way that local variables can be lifted to generated class fields
+					//value of "this" comes here either
+					if (res_name != undefined && res_name.indexOf ('>') > 0) {
+						// For async methods, we get the names too, so use that
+						// ALTHOUGH, the name wouldn't have `<>` for method args
+						res [i].name = res_name.substring (1, res_name.indexOf ('>'));
+					}
 
-			if (this._async_method_objectId != 0) {
-				for (let i in res) {
-					if (res [i].value.isValueType != undefined && res [i].value.isValueType)
-						res [i].value.objectId = `dotnet:valuetype:${this._async_method_objectId}:${res [i].fieldOffset}`;
+					if (value.isValueType != undefined && value.isValueType)
+						value.objectId = `dotnet:valuetype:${this._async_method_objectId}:${res [i].fieldOffset}`;
+				} else if (res_name === undefined && var_list [i] != undefined) {
+					// For non-async methods, we just have the var id, but we have the name
+					// from the caller
+					res [i].name = var_list [i][1];
```

Comment:
not going to demand this be fixed, but the use of arrays-as-nameless-tuples like this in JS is a smell that usually leads to buggy code. if possible it would be much better for them to be objects with named properties, the cost isn't worse and it's much easier to debug